src/main/java/rx/Single.java

```diff
@@ -1801,6 +1804,9 @@ public final void onError(Throwable e) {
             public final void onSuccess(T args) {
                 try {
                     onSuccess.call(args);
+                } catch (Throwable ex) {
+                    Exceptions.throwIfFatal(ex);
+                    RxJavaHooks.onError(ex);
```

Comment:
Hm, that's not what Observable does in 1.x, I think we should call `Exceptions.throwOrReport()`, in 1.x we didn't call hooks/plugins in such cases

See 

https://github.com/ReactiveX/RxJava/blob/5b2394c9ee91f298661fff5e043744c84b425808/src/main/java/rx/observers/SafeSubscriber.java#L139

https://github.com/ReactiveX/RxJava/blob/5b2394c9ee91f298661fff5e043744c84b425808/src/main/java/rx/exceptions/Exceptions.java#L222

Reply:
Calling that would just bring back the problematic behavior, i.e., calling `onError` when `onSuccess` crashes.